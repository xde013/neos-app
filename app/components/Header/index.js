import React from 'react';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';

function Header() {
  return (
    <Grid container direction="row" justify="center" alignItems="center">
      <Box p={2}>
        <Button
          data-testid="home"
          variant="outlined"
          to="/"
          component={Link}
          color="primary"
          size="large"
        >
          Home
        </Button>
      </Box>

      <Box p={2}>
        <Button
          data-testid="about"
          variant="outlined"
          to="/about"
          component={Link}
          color="primary"
          size="large"
        >
          About
        </Button>
      </Box>
    </Grid>
  );
}

export default Header;
