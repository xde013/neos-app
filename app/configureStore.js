/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createSagaMiddleware from 'redux-saga';
import createReducer from './reducers';

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({});
    /* eslint-enable */
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with two middlewares
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  // Hot Reloadable Reducers
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers));
    });
  }

  return store;
}
