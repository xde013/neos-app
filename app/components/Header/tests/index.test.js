import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';

import Header from '../index';
import configureStore from '../../../configureStore';

describe('<Header />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should match snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Header />
        </ConnectedRouter>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have link to Home page', () => {
    const { queryByTestId } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Header />
        </ConnectedRouter>
      </Provider>,
    );

    expect(queryByTestId(/home/i)).toBeTruthy();
  });

  it('should have link to About page', () => {
    const { queryByTestId } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Header />
        </ConnectedRouter>
      </Provider>,
    );

    expect(queryByTestId(/about/)).toBeTruthy();
  });
});
