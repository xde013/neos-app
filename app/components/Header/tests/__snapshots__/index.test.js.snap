// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiBox-root MuiBox-root-105"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
      data-testid="home"
      href="/"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        Home
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiBox-root MuiBox-root-138"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
      data-testid="about"
      href="/about"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        About
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
